test/test_implicit_free_surface_solver.jl

```diff
@@ -0,0 +1,101 @@
+using Statistics
+using Oceananigans.Buoyancy: g_Earth
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: ImplicitFreeSurface, FreeSurface, 
+      compute_volume_scaled_divergence!, add_previous_free_surface_contribution,
+      compute_vertically_integrated_volume_flux!
+
+function run_implicit_free_surface_solver_tests(arch)
+    Nx = 360
+    Ny = 360
```

Comment:
Can we reduce the resolution for this test or is this required?

Reply:
It only does one solve, so the size doesn't seem to slow it down.